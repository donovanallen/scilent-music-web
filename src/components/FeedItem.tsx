'use client';

import { Track } from '@spotify/web-api-ts-sdk';
import React from 'react';
import { BiAlbum } from 'react-icons/bi';
import { GiBackwardTime } from 'react-icons/gi';

import { cn, formatArtists, getTimestampText } from '@/lib/utils';

import NextImage from '@/components/NextImage';

interface FeedItemProps {
  data: Track;
  timestamp?: Date;
  onClick?: (id: string) => void;
  className?: string;
}

const FeedItem: React.FC<FeedItemProps> = ({
  data,
  timestamp,
  onClick,
  className,
}) => {
  const handleClick = () => {
    if (onClick) {
      return onClick(data.id);
    }
  };

  return (
    <div
      onClick={handleClick}
      className={cn(
        'flex items-start gap-x-2 cursor-pointer w-full rounded-md hover:bg-neutral-800/50',
        className,
      )}
    >
      <div className='relative rounded-md min-h-[48px] min-w-[48px] overflow-hidden aspect-square m-1 bg-neutral-700'>
        {data.album?.images ? (
          <NextImage
            src={data.album?.images[0].url}
            alt={`Album image: ${data.name}`}
            layout='fill'
            className='aspect-square object-cover'
            useSkeleton
          />
        ) : (
          <BiAlbum size={24} className='m-auto h-full text-dark' />
        )}
      </div>
      <div className='flex flex-col overflow-hidden'>
        <p className='text-light truncate'>{data.name}</p>

        {data.artists && (
          <p className='text-neutral-400 subtitle truncate'>
            {formatArtists(data.artists)}
          </p>
        )}

        {timestamp && (
          <div className='inline-flex gap-x-1 items-center text-neutral-500 truncate'>
            <GiBackwardTime fontSize='small' />
            <span className='subtitle text-xs'>
              {getTimestampText(timestamp.toString())}
            </span>
          </div>
        )}
      </div>
    </div>
  );
};

export default FeedItem;
