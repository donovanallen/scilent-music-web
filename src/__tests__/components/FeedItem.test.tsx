import { Track } from '@spotify/web-api-ts-sdk';
import { render } from '@testing-library/react';
import React from 'react';

import FeedItem from '@/components/FeedItem';

describe('FeedItem', () => {
  const mockData = {} as Track;
  it('renders correctly', () => {
    const { asFragment } = render(<FeedItem data={mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
