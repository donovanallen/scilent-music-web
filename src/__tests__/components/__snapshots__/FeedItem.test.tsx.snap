// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-start gap-x-2 cursor-pointer w-full rounded-md hover:bg-neutral-800/50"
  >
    <div
      class="relative rounded-md min-h-[48px] min-w-[48px] overflow-hidden aspect-square m-1 bg-neutral-700"
    >
      <svg
        class="m-auto h-full text-dark"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11.99"
          cy="11.99"
          r="2.01"
        />
        <path
          d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
        />
        <path
          d="M12 6a6 6 0 0 0-6 6h2a4 4 0 0 1 4-4z"
        />
      </svg>
    </div>
    <div
      class="flex flex-col overflow-hidden"
    >
      <p
        class="text-light truncate"
      />
    </div>
  </div>
</DocumentFragment>
`;
